tests/MnsQueueTest.php

```diff
@@ -31,141 +27,122 @@
-        ]);
-    }));
-
-    $this->mockedMessageNotExistsResult = new ReceiveMessageResult($this->mockedNotFoundResponse, tap(Mockery::mock(XmlEncoder::class), function ($mock) {
-        $mock->expects()->decode('<response></response>')->andReturns([
+            'MessageBodyMD5' => hash('md5', $this->mockedPayload),
+        ],
+    ]));
+
+    $this->mockedReceiveMessage = [
+        'MessageId' => $this->mockedMessageId,
+        'MessageBody' => $this->mockedPayload,
+        'MessageBodyMD5' => hash('md5', $this->mockedPayload),
+        'ReceiptHandle' => $this->mockedReceiptHandle,
+        'EnqueueTime' => '1250700979248',
+        'NextVisibleTime' => '1250700799348',
+        'FirstDequeueTime' => '1250700779318',
+        'DequeueCount' => '1',
+        'Priority' => '8',
+    ];
+
+    $this->mockedReceiveMessageResult = new ReceiveMessageResult(new Result([
+        'Message' => $this->mockedReceiveMessage,
+    ]));
+
+    $this->mockedMessageNotExistsException = QueueException::makeFromResult(new Result([
+        'Error' => [
             'Code' => 'MessageNotExist',
             'Message' => 'Message not exist.',
-        ]);
-    }));
-
-    $this->mockedBatchReceiveMessageResult = new BatchReceiveMessageResult($this->mockedOkResponse, tap(Mockery::mock(XmlEncoder::class), function ($mock) {
-        $mock->expects()->decode('<response></response>')->andReturns([
-            'Message' => [
-                ['ReceiptHandle' => $this->mockedReceiptHandle],
-            ],
-        ]);
-    }));
-
-    $this->mockedBatchMessageNotFoundResult = new BatchReceiveMessageResult($this->mockedNotFoundResponse, tap(Mockery::mock(XmlEncoder::class), function ($mock) {
-        $mock->expects()->decode('<response></response>')->andReturns([
+        ],
+    ]));
+
+    $this->mockedBatchReceiveMessageResult = new BatchReceiveMessageResult(new Result([
+        'Messages' => ['Message' => [$this->mockedReceiveMessage]],
+    ]));
+
+    $this->mockedBatchMessageNotFoundException = QueueException::makeFromResult(new Result([
+        'Error' => [
             'Code' => 'MessageNotExist',
             'Message' => 'Message not exist.',
-        ]);
-    }));
+        ],
+    ]));
 
-    $this->mockedBatchDeleteMessageResult = new BatchDeleteMessageResult($this->mockedNoContentResponse, Mockery::mock(XmlEncoder::class));
+    $this->mockedBatchDeleteMessageResult = new MnsResult(new Result);
 });
 
 test('queue size includes active, inactive and delayed messages', function () {
-    $this->mns->expects()->getQueueAttributes($this->queueName)->andReturns($this->mockedGetQueueAttributesResult);
-    $queue = new MnsQueue($this->mns, $this->queueName);
-    expect($queue->size())->toBeInt()->toBe($this->mockedActiveMessages + $this->mockedInactiveMessages + $this->mockedDelayMessages);
+    $this->console->shouldReceive('getQueueAttributes')->with(['QueueName' => $this->queueName])->once()->andReturn($this->mockedGetQueueAttributesResult);
```

Comment:
We can use `expects()` to refactor some `shouldReceive()` statements to make them a bit shorter.